[JSC] Remove `--useIteratorHelpers` flag

sosukesuzuki opened the pull request:
#### 94489ff12ce643dbcb1c84249627a97bccea858b
<pre>
[JSC] Remove `--useIteratorHelpers` flag
<a href="https://bugs.webkit.org/show_bug.cgi?id=292526">https://bugs.webkit.org/show_bug.cgi?id=292526</a>

Reviewed by Yusuke Suzuki.

Iterator Helpers Proposal[1] is now stage 4 and enabled by default in
JSC. So this patch removes `--useIteratorHelpers` flag.

* JSTests/stress/iterator-concat.js:
* JSTests/stress/iterator-from.js:
* JSTests/stress/iterator-prototype-chunks.js:
* JSTests/stress/iterator-prototype-constructor-new.js:
* JSTests/stress/iterator-prototype-constructor-setter.js:
* JSTests/stress/iterator-prototype-every.js:
* JSTests/stress/iterator-prototype-find.js:
* JSTests/stress/iterator-prototype-forEach.js:
* JSTests/stress/iterator-prototype-map.js:
* JSTests/stress/iterator-prototype-reduce.js:
* JSTests/stress/iterator-prototype-some.js:
* JSTests/stress/iterator-prototype-to-string-tag-basic-setter.js:
* JSTests/stress/iterator-prototype-toArray.js:
* JSTests/stress/iterator-prototype-windows.js:
* JSTests/test262/config.yaml:
* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSGlobalObject::init):
* Source/JavaScriptCore/runtime/JSIteratorConstructor.cpp:
(JSC::JSIteratorConstructor::finishCreation):
* Source/JavaScriptCore/runtime/JSIteratorPrototype.cpp:
(JSC::JSIteratorPrototype::finishCreation):
* Source/JavaScriptCore/runtime/OptionsList.h:

Canonical link: <a href="https://commits.webkit.org/294591@main">https://commits.webkit.org/294591@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7b7ea02a85a4cd3b6a7c9e53477f1a273db29cf1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/102102 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21770 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/12086 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/107261 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52738 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/22079 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30277 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77712 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34706 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/105109 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/17084 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/92181 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58047 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16912 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10206 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/52096 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/94774 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86752 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10279 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109636 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/100712 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/29235 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21553 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86683 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29597 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88385 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86265 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21998 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31067 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8786 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/23460 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/29163 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34458 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/124338 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28974 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/34528 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32297 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30533 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

justinmichaud reviewed (commented):
LGTM, but I'll wait for someone from Apple to approve

Constellation reviewed (approved):
r=me

sosukesuzuki committed:
[JSC] Remove `--useIteratorHelpers` flag
https://bugs.webkit.org/show_bug.cgi?id=292526

Reviewed by Yusuke Suzuki.

Iterator Helpers Proposal[1] is now stage 4 and enabled by default in
JSC. So this patch removes `--useIteratorHelpers` flag.

* JSTests/stress/iterator-concat.js:
* JSTests/stress/iterator-from.js:
* JSTests/stress/iterator-prototype-chunks.js:
* JSTests/stress/iterator-prototype-constructor-new.js:
* JSTests/stress/iterator-prototype-constructor-setter.js:
* JSTests/stress/iterator-prototype-every.js:
* JSTests/stress/iterator-prototype-find.js:
* JSTests/stress/iterator-prototype-forEach.js:
* JSTests/stress/iterator-prototype-map.js:
* JSTests/stress/iterator-prototype-reduce.js:
* JSTests/stress/iterator-prototype-some.js:
* JSTests/stress/iterator-prototype-to-string-tag-basic-setter.js:
* JSTests/stress/iterator-prototype-toArray.js:
* JSTests/stress/iterator-prototype-windows.js:
* JSTests/test262/config.yaml:
* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSGlobalObject::init):
* Source/JavaScriptCore/runtime/JSIteratorConstructor.cpp:
(JSC::JSIteratorConstructor::finishCreation):
* Source/JavaScriptCore/runtime/JSIteratorPrototype.cpp:
(JSC::JSIteratorPrototype::finishCreation):
* Source/JavaScriptCore/runtime/OptionsList.h:

Canonical link: https://commits.webkit.org/294591@main
